### PR TITLE
MINOR: Add test cases to the Raft module

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds a few test cases to the Raft module (i.e., VotedStateTest, ValidOffsetAndEpochTest, UnattachedStateTest, ResignedStateTest). I have also removed an un-used method (i.e., `overrideElectionTimeout`) from VotedState class.

Signed-off-by: morsak <maros.orsak159@gmail.com>

### CODE REVIEW

File: raft/src/main/java/org/apache/kafka/raft/VotedState.java
Comment (divijvaidya) [inline raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: I understand that it is an unused method but removing or adding public methods requires a KIP in the Kafka community.

Comment (see-quick) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: I will revert this method deletion and let only these test cases. Sounds good? I wonder if adding a test case for such an (un-used) method would be beneficial...

Comment (mimaison) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: `VotedState` is not part of the public API so we can change this public method without a KIP.

Comment (divijvaidya) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: ok, I stand corrected. Apologies for the confusion here @see-quick. (note to myself) Here is the guidance that we can follow for what requires a KIP for future: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals#KafkaImprovementProposals-Whatisconsidereda%22majorchange%22thatneedsaKIP?

Comment (see-quick) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: > ok, I stand corrected. Apologies for the confusion here @see-quick. (note to myself) Here is the guidance that we can follow for what requires a KIP for future: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals#KafkaImprovementProposals-Whatisconsidereda%22majorchange%22thatneedsaKIP?

Thanks for sharing that link @divijvaidya <3

Comment (mimaison) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: You can find the list of public classes/interfaces in the javadocs: https://kafka.apache.org/32/javadoc/allclasses-index.html

Comment (see-quick) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: Thanks noted.

Comment (see-quick) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: So basically we can merge it or? :) ^^ @divijvaidya ?

Comment (divijvaidya) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: Yes code looks good from my side 👍

We need to re-run the tests since the build had a bunch of failures due to a lingering thread. We expect this commit https://github.com/apache/kafka/commit/09011da76d450f9a7dd11e159741f8e69cca115b to mitigate the lingering thread problem and unblock the build. Please check if you have this commit in your repo. If not, I would suggest to rebase this test with `trunk` and run again. If you already have this commit, then we need to ask a committer to re-run the tests.

Comment (see-quick) [reply raft/src/main/java/org/apache/kafka/raft/VotedState.java:95]: I have got that commit in my local repo, so as you propose we need to ask a committer to re-run the tests :)

**Comment by ijuma** [comment]
Thanks for the PR. I don't understand the goal of the `toString` tests since they basically check that the toString doesn't have `@`?

**Comment by see-quick** [comment]
> Thanks for the PR. I don't understand the goal of the `toString` tests since they basically check that the toString doesn't have `@`?

The main purpose of such test cases is to check that the `toString` method does print something useful. Meaning not printing object hash code from the inherited superclass (i.e., `java.lang.Object@7fbe847c`). If you do not see any value in such test cases I can remove them.

**Comment by see-quick** [comment]
^^ @ijuma so WDYT? :)

**Review by mimaison — APPROVED** [review]
LGTM Thanks for the PR!

**Pull request merged**
